src/utils/__tests__/webcrypto.test.js

```diff
@@ -4,6 +4,16 @@ import { digest, getRandomValues } from '../webcrypto';
 jest.mock('js-sha1');
 
 describe('util/webcrypto', () => {
+    beforeEach(() => {
+        // eslint-disable-next-line no-undef
+        Object.defineProperty(globalThis, 'window', {
+            value: {
+                ...window,
```

Comment:
when the `window` object is modified in one of the tests below, doesn't this set the modified values back to the window object? shouldn't we be resetting the window object back to the original state before each test?

and/or mocking only the `crypto` and `msCrypto` properties so the rest of the object is confidently unchanged

Reply:
with the `beforeEach` each test will have a copy of the same base object which is just a writable `window` object, if the test changes it, it only affects the one that is in that test. The key here is that `window` is read only for some reason, so I had to replace the whole `window` object with a writable `window` object.